dev/cli

```diff
@@ -83,6 +161,37 @@ _get_gcp_token_payload() {
   echo "$decoded_gcp_token_payload"
 }
 
+function fetch_gcf_tokens() {
+  echo "-- ------------------------------------------------------ --"
+  echo "-- Generate Google Cloud GCP Identity tokens from GCF URL --"
+  echo "-- ------------------------------------------------------ --"
+
+  local gcp_func_url=$1
+  local token_prefix="tokens/gcf_"
+
+  validate_gcf_url_accessible $gcp_func_url
+
+  cd ../ci/authn-gcp/
+  sh ./get_tokens_to_files.sh "$gcp_func_url" "$token_prefix" || exit 1
+  cd -
+
+  echo "Finished obtaining and writing tokens to file"
+}
+
+function validate_gcf_url_accessible() {
+  echo "Validating google cloud function is accessible"
+
+  local gcp_func_url=$1
+  local audience="dummy_host"
+  local status_code=$(curl -o /dev/null -s -w "%{http_code}\n" \
+    -H 'Metadata-Flavor: Google' "$gcp_func_url?audience=$audience")
+
+  if [ ! "$status_code" = "200" ]; then
+    echo "Failed to access function, HTTP Status: $status_code"
+    exit 1
+  fi
+}
+
```

Comment:
source the original script and call the function.

Reply:
source is actually running the script, i extracted the logic to another script

Reply:
because there's  a call to a function at the bottom of the script.